File: treeorders.c
#include <stdio.h>
#include <stdlib.h>
int number = 15;
typedef struct treeNode {
    int data;
    struct treeNode *leftChild;
    struct treeNode *rightChild;
}tree;
tree *selector;

void preOrder(tree *selector){
    if(selector){
    printf("%d ", selector->data);
    preOrder(selector->leftChild);
    preOrder(selector->rightChild);
    }
}

void inOrder(tree *selector){
    if(selector){
    inOrder(selector->leftChild);
    printf("%d ", selector->data);
    inOrder(selector->rightChild);
    }
}

void postOrder(tree *selector){
    if(selector){
    postOrder(selector->leftChild);
    postOrder(selector->rightChild);
    printf("%d ", selector->data);
    }
}

int main(int argc, char *argv){
    /*tree arr[10];
    tree nodes[number + 1];
	for(int i = 1; i <= number; i++) {
		nodes[i].data = i;
		nodes[i].leftChild = NULL;
		nodes[i].rightChild = NULL;
	}
	for(int i = 1; i <= number; i++) {
		if(i % 2 == 0)
			nodes[i / 2].leftChild = &nodes[i];
		else
			nodes[i / 2].rightChild = &nodes[i];
	}
	preOrder(&nodes[1]);
	return 0;*/
    tree *arr[10];
    for(int i = 0; i < 10; i++){
        tree *temp = malloc(sizeof(tree));
        temp->data = i;
        temp->leftChild = NULL;
        temp->rightChild = NULL;
        arr[i] = temp;
    }
    for(int i = 0; i < 10; i++){
        if(i % 2 == 0) arr[i/2]->leftChild = arr[i];
        else arr[i/2]->rightChild = arr[i];
    }

    preOrder(arr[1]);
}

File: treeorders/treeorders.js
// Tree 
class Tree {
    constructor(x){
        this.data=x;
        this.leftChild=null;
        this.rightChild=null;
    }
}

function preOrder(selector){
    if(selector != null){
        console.log(selector.data);
        preOrder(selector.leftChild);
        preOrder(selector.rightChild);
    }
}

function inOrder(selector){
    if(selector != null){
        inOrder(selector.leftChild);
        console.log(selector.data);
        inOrder(selector.rightChild);
    }
}

function postOrder(selector){
    if(selector != null){
        postOrder(selector.leftChild);
        postOrder(selector.rightChild);
        console.log(selector.data);
    }
}

arr = [1,2,3,4,5,6,7,8,9, 10];
for(let i = 0; i < arr.length; i++){
    let tre = new Tree(i);
    arr[i] = tre;
}

for(let i = 0; i < arr.length; i++){
    if(i % 2 == 0) arr[i/2].leftChild = arr[i];
    else arr[parseInt(i/2)].rightChild = arr[i];
}

preOrder(arr[1]);